flink-runtime/src/main/java/org/apache/flink/runtime/messages/webmonitor/JobDetails.java

```diff
@@ -411,7 +382,26 @@ public JobDetails deserialize(
                     lastUpdateTime,
                     numVerticesPerExecutionState,
                     numTasks,
-                    attempts);
+                    new HashMap<>());
+        }
+    }
+
+    public static final class CurrentAttempts implements Serializable {
+        private final int representativeAttempt;
+
+        private final List<Integer> currentAttempts;
```

Comment:
Sorry for not think about it initially, but we might change the data structure to Set<Integer> since the only usage of this field is to query existence.

Reply:
I've been thinking about this but I was wondering if it's too customized for this pr. But maybe it's ok. I'll modify here.